### DISCUSSION TITLE
Looking at Sentinel data through the Copernicus browser

### DISCUSSION THREAD

**Discussion by ipoole** [SimonFisher92/Scottish_Snow_Satellite_Data#22] [General]
The Copernicucus.eu data browser is nice, I'm just starting to explore it.   (I was pointed at it by an error message when using our download code, a command which recently worked, has there been a change in policy by the Copernicus team?).  This seems quite powerful, I wasn't previously aware of it.

The following should show the Cairngorms in May this year.

https://dataspace.copernicus.eu/browser/?zoom=13&lat=57.07191&lng=-3.61038&themeId=DEFAULT-THEME&visualizationUrl=https%3A%2F%2Fsh.dataspace.copernicus.eu%2Fogc%2Fwms%2Fa91f72b5-f393-4320-bc0f-990129bd9e63&datasetId=S2_L2A_CDAS&fromTime=2023-05-30T00%3A00%3A00.000Z&toTime=2023-05-30T23%3A59%3A59.999Z&layerId=SCENE-CLASSIFICATION&demSource3D=%22MAPZEN%22&cloudCoverage=30

**Comment by murraycutforth** [comment]
Yeah that is quite nice, maybe this already offers all the functionality we need! Have you seen this @EddieBoyle2019 ? I think the only disadvantage is that if you want to disambiguate cloud/snow then you have to use the SCL band, which I think is pretty rubbish. It's also not clear exactly what date the image which you're looking at was taken at, since it seems to stitch together images (using either a most recent or least cloudy mosaicking rule), but maybe that can be addressed somewhere in their interface?

**Comment by ipoole** [comment]
Yes I think it is quite powerful.   Is this a new resource does anyone know?

Concerning dates, I think you can specify a specific data (from 2018 onwards at least), with the coverage (zoom out) reflecting the flyover for that day.   You can also filter the offered dates (in the calendar) by cloud cover percentage.

The compare facility is nice.  The following (do these links work?) compares 16th May 2018 with 30th May 2023, showing the 'scene classification' layer.   You can clearly see the reduced snow coverage.   Or you can select the true colour layer, confirming the result.   Btw, there is a note saying you need to login to access 'advanced features'.   I created an account yesterday.

https://dataspace.copernicus.eu/browser/?zoom=13&lat=57.06762&lng=-3.6521&themeId=DEFAULT-THEME&visualizationUrl=https%3A%2F%2Fsh.dataspace.copernicus.eu%2Fogc%2Fwms%2Fa91f72b5-f393-4320-bc0f-990129bd9e63&datasetId=S2_L2A_CDAS&fromTime=2018-05-16T00%3A00%3A00.000Z&toTime=2018-05-16T23%3A59%3A59.999Z&layerId=SCENE-CLASSIFICATION&demSource3D=%22MAPZEN%22&cloudCoverage=14.

There is also a time lapse facility you can configure ... looking at that.

**Comment by EddieBoyle2019** [comment]
Hello everyone, just trying to catch up on things, this Copernicus browser looks fantastic.

I have not seen it before, but it looks like it's been available since the summer:
https://dataspace.copernicus.eu/news/2023-4-20-discover-our-new-services

Particularly interested to see that they have published the technical details of the algorithm used to generate the Sentinel-2 L2A Scene Classification Map:

https://custom-scripts.sentinel-hub.com/sentinel-2/scene-classification/
https://sentinels.copernicus.eu/web/sentinel/technical-guides/sentinel-2-msi/level-2a/algorithm-overview

The snow or ice features showing up as easy-to-see pink features on the classification map is great. The 20m resolution of the classification map is not as good as the 10m of the true colour map, and misses the smaller patches, but I still think it will offer a great tool for snowpatch researchers/observers.

What I would say is that Murray's Streamlit app offers these advantages over the Copernicus browser:

- Better background map (contours etc)
- Easy to use intuitive interface
- Narrow focus, just shows exactly the locations we need (drop-down menu etc)
- Comparison overlay of the different layers in a small window of the observation area

If one us could build a new version of this app which includes the classification map, then we've pretty much got the ideal tool right there (IMHO)!

**Reply by ipoole** [reply]
Thanks Simon for those links.   Yes the scene classification algorithm description is very thorough.   It seems to be highly hand crafted, old-school image analysis (like wot I used to do...) , embodying deep understanding of the Sentinel data and atmospheric physics, with many manually set thresholds.  It looks like many man-years of effort which we would struggle to improve on.

I looked again at Murray's Streamlit app, and indeed it is very nice, minimal, streamlined  and under our control.   I should better understand how/where the data is stored and the Streamlit architecture...

**Reply by EddieBoyle2019** [reply]
Yes, this algorithm is old school! No machine leaning there, so I guess there is still room for a potential solution involving CNNs, but as you say, a big question is whether it can improve upon this existing algorithm.

**Comment by murraycutforth** [comment]
Also @EddieBoyle2019 - I've just noticed that there is a contour option built into the copernicus browser:

![image](https://github.com/SimonFisher92/Scottish_Snow/assets/11088372/434f0431-d2dc-40d8-b58f-292d4287989d)

You hover over the "layers" box in the top right corner, and select it from the menu which appears.
